back/src/controllers/products/GetProductById.go

```diff
@@ -10,12 +10,12 @@ import (
 func GetProductByID(ctx *fiber.Ctx) error {
 	var product model.Product
 
-	if err := SelectProductByID(&product, ctx); err != nil {
-		return helpers.SetStatusAndMessages(
-			constants.InternalServerErrorStatus,
-			constants.InternalServerErrorMessage,
-			ctx)
+	if err := SelectProductByID(&product, ctx); err == nil {
+		return sendProduct(ctx, product)
 	}
 
-	return sendProduct(ctx, product)
```

Comment:
Do not rework this, we need to keep sending the product.

Reply:
We can still send products I just changed the condition but the behaviour of the code should be the same

Reply:
It creates a slight coding style change, I prefer that we don't change it if you agree.

Reply:
Done

Reply:
Thanks